A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -705,8 +705,8 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
 
         //Changes to climb phase when acceleration altitude is reached
         if (this.currentFlightPhase === FlightPhase.FLIGHT_PHASE_TAKEOFF && airSpeed > 80) {
-            const agl = Simplane.getAltitudeAboveGround();
-            if (agl > (this.accelerationAltitude || this.thrustReductionAltitude || 1500)) {
+            const msl = Simplane.getAltitude();
+            if (msl > (this.accelerationAltitude || this.thrustReductionAltitude || 1500)) {
```

Comment:
If we are using MSL, the default `1500` will not work at airports higher than 1500 ft. This needs to be changed to airport elevation + 1500. We should also check to make sure acceleration altitude and thrust reduction altitude are above the airport elevation.

Reply:
Agreed. Only thing I'm not sure on right now is where the default thr red/acc is set. Could you perhaps point me in the correct direction for that? Then I will work on it as soon as I can.

Reply:
Okay I just realized I interpreted this completely wrong probably. I was assuming you were talking about the default value on the MCDU page. While I still think that should be changed, and I'd be open to change it, I'll at least change the default value in the check when I get out of work tomorrow.

MCDU change might be better in a different issue anyways, since it seems there are still some discrepancies if I look at something like [http://www.a320dp.com/A320_DP/nav-flight-management/sys-14.5.3.html](http://www.a320dp.com/A320_DP/nav-flight-management/sys-14.5.3.html)

Reply:
Should the default be tied to airport? Or just current ground level?

```
        if (this.currentFlightPhase === FlightPhase.FLIGHT_PHASE_TAKEOFF && airSpeed > 80) {
            const agl = Simplane.getAltitudeAboveGround();
            const msl = Simplane.getAltitude();
            const heightAboveSeaLevel = msl - agl;
            if (msl > (this.accelerationAltitude || this.thrustReductionAltitude || (1500 + heightAboveSeaLevel))) {
                this.currentFlightPhase = FlightPhase.FLIGHT_PHASE_CLIMB;
            }
        }
```

Or maybe something else. :)

Reply:
On a second note, I don't think you can even have both as unset, i.e. null. Game automatically sets one of them and you cannot clear it as far as I can tell, so the third condition does not ever affect in practise. Doesn't hurt to have something there though.

Reply:
Yeah exactly. There is actually a simvar containing the current MSL altitude of the ground, so I'm going to use that. Commit coming up in a bit after I've tested it.

Reply:
I think we should use the airport altitude, not the current ground altitude.

Reply:
Good point. I got the information I need for it now. I'll change it here and in #585

Reply:
I kept the ground altitude in there, but only if there is no origin airport selected in the CDU.

Reply:
If the ground isn't flat, that could cause problems. I think it would be better if we used the initial ground altitude after takeoff.

Reply:
I wouldn't spend too much time on the default value. As said, it will likely never be executed as you cannot unset both THR RED/ACC, you always have one (or soon two, like in real aircraft). It would be more important to get the actual fix in place.

Reply:
You might not always have those values after I implement my other PR. If you start a flight without a destination selected, your origin airport on the INIT A page will be empty, and then so will the THR RED/ACC values. Without a default value this could mean that you get stuck in Take Off mode. This will be a small change anyways.

Reply:
Thanks for clarification.